src/utils.rs

```diff
@@ -229,4 +229,17 @@ mod tests {
         check_setenv("Test");
         check_setenv("Тест"); // "Test" in Russian
     }
+
+    #[test]
+    fn test_filename_from_uri() {
+        use gstring::GString;
+        use std::path::PathBuf;
+        let uri: GString = "file:///foo/bar.txt".into();
```

Comment:
Maybe also add a test with a hostname for completeness?

Reply:
In which case can hostname be non-null? I tried an http uri but:
`Error { domain: g_convert_error, code: 4, message: "The URI “http://gtk-rs.org/index.html” is not an absolute URI using the “file” scheme" }`

Reply:
I don't know :) Maybe check the code of the C function?

Reply:
Yes the PR is updated now :)